src/ws-dropdown/dropdown-input.js

```diff
@@ -0,0 +1,115 @@
+import {React, Component} from '../imports';
+
+const KEY_ENTER = 13;
+
+/**
+ * Renders renders a text input inside a dropdown container to provide a free text dropdown component.
+ * Possible options are value and placeholder which are both strings.
+ */
+export class DropdownInput extends Component {
+  /**
+   * @type {Object}
+   */
+  static defaultProps = {
+    value: null,
+    placeholder: '',
+    handle: () => {}
+  };
+
+  /**
+   * @type {Object}
+   */
+  static propTypes = {
+    value: React.PropTypes.string,
+    placeholder: React.PropTypes.string,
+    handle: React.PropTypes.func
+  };
+
+  /**
+   * @param {Object} props React props
```

Comment:
Please add a definition about PropTypes (see WS-Header). They are not auto generated.